pages/index.tsx

```diff
@@ -48,69 +52,90 @@ export default function Home() {
+              )
+              .trim(),
+            percentage: toxicity,
+            suggestion: "Kimi",
+          });
+        }
       }
+      // get rid of duplicate texts
+      var out = temp.reduce(function (p: any, c: any) {
+        if (
+          !p.some(function (el: any) {
+            return el.text === c.text;
+          })
+        )
+          p.push(c);
+        return p;
+      }, []);
+      // sort by percentage
+      out.sort(function (left: SentenceAndScore, right: SentenceAndScore): number {
+        if (left.percentage < right.percentage) {
+          return 1;
+        }
+        if (left.percentage > right.percentage) {
+          return -1;
+        }
+        return 0;
+      });
+      setSentencesAndScores(out);
     }
-    // get rid of duplicate texts
-    var out = temp.reduce(function (p: any, c: any) {
-      if (
-        !p.some(function (el: any) {
-          return el.text === c.text;
-        })
-      )
-        p.push(c);
-      return p;
-    }, []);
-    // sort by percentage
-    out.sort(function (left: SentenceAndScore, right: SentenceAndScore): number {
-      if (left.percentage < right.percentage) {
-        return 1;
-      }
-      if (left.percentage > right.percentage) {
-        return -1;
-      }
-      return 0;
-    });
-    setSentencesAndScores(out);
   }
 
   function editInputText(original: string, suggestion: string) {
-    console.log(original);
-    console.log(suggestion);
     let temp: string = userInput.replace(original, suggestion);
     setUserInput(temp);
   }
 
-  // #FIXME: Replace with stateful operations
+  function getWeightedScores(toxicScore: number, profaneScore: number, insultScore: number, threatScore: number): number {
+    var sumWeight = 0, out = 0;
+    sumWeight += scoreCategoriesSettings[ScoreCategory.toxic].enabled ? scoreCategoriesSettings[ScoreCategory.toxic].weight : 0; 
+    sumWeight += scoreCategoriesSettings[ScoreCategory.profane].enabled ? scoreCategoriesSettings[ScoreCategory.profane].weight : 0; 
+    sumWeight += scoreCategoriesSettings[ScoreCategory.insult].enabled ? scoreCategoriesSettings[ScoreCategory.insult].weight : 0; 
+    sumWeight += scoreCategoriesSettings[ScoreCategory.threat].enabled ? scoreCategoriesSettings[ScoreCategory.threat].weight : 0; 
+    out += scoreCategoriesSettings[ScoreCategory.toxic].enabled ? toxicScore * (scoreCategoriesSettings[ScoreCategory.toxic].weight / sumWeight) : 0;
+    out += scoreCategoriesSettings[ScoreCategory.profane].enabled ? profaneScore * (scoreCategoriesSettings[ScoreCategory.profane].weight / sumWeight) : 0;
+    out += scoreCategoriesSettings[ScoreCategory.insult].enabled ? insultScore * (scoreCategoriesSettings[ScoreCategory.insult].weight / sumWeight) : 0;
+    out += scoreCategoriesSettings[ScoreCategory.threat].enabled ? threatScore * (scoreCategoriesSettings[ScoreCategory.threat].weight / sumWeight) : 0;
+    return out;
+  }
+
   const getPercentage = () => {
```

Comment:
The `getPercentage` function is no longer being used, b/c it’s being automatically calculated in (#41).